From 7b9e9d87bccbdfa2e1581de56cad4920961af899 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Wed, 7 Oct 2020 16:40:24 +0300
Subject: [PATCH 29/76] nvme: Workaround Samsung 970 Pro power state issues

Sometimes Samsung 970/960 Pro NVME device just disappears.
Looks like it drops into deep sleep mode during I/O.
This behaviour has been observed at least on R-Car
Gen3 H3ULCB and M3ULC Kingfisher boards.

Add a quirk that disables deep sleep mode for the device.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/nvme/host/pci.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index c1f34462..c25b722 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -3244,6 +3244,10 @@ static const struct pci_device_id nvme_id_table[] = {
 		.driver_data = NVME_QUIRK_DELAY_BEFORE_CHK_RDY, },
 	{ PCI_DEVICE(0x1c5f, 0x0540),	/* Memblaze Pblaze4 adapter */
 		.driver_data = NVME_QUIRK_DELAY_BEFORE_CHK_RDY, },
+	{ PCI_DEVICE(0x144d, 0xa804),	/* Samsung 960 Pro */
+		.driver_data = NVME_QUIRK_NO_DEEPEST_PS },
+	{ PCI_DEVICE(0x144d, 0xa808),	/* Samsung 970 Pro */
+		.driver_data = NVME_QUIRK_NO_DEEPEST_PS },
 	{ PCI_DEVICE(0x144d, 0xa821),   /* Samsung PM1725 */
 		.driver_data = NVME_QUIRK_DELAY_BEFORE_CHK_RDY, },
 	{ PCI_DEVICE(0x144d, 0xa822),   /* Samsung PM1725a */
-- 
2.7.4

